### PR TITLE
Update dependencies, fix nan deprecation warnings

### PR DESCRIPTION
**Pull request by MayhemYDG**
Additionally:
I fixed a bunch of unused variables/dependencies, you should setup the repo's `.jshintrc` to catch those.
`async` is used in `bin/zopfli` so I moved it into `dependencies `.
I also fiddled with some imports in the C code, it still compiles and tests pass so I assume everything is alright.

### CODE REVIEW

**Comment by jakubpawlowicz** [comment]
:+1: I had to manually install async because it's not listed in dependencies.

**Comment by pierreinglebert** [comment]
I forgot about this PR, will do it tomorrow and publish :)

**Comment by jakubpawlowicz** [comment]
:+1: thanks!

File: src/png/zopflipng.cc [hunk 1]
```diff
@@ -1,14 +1,10 @@
-#include <node.h>
```
Comment (pierreinglebert) [inline src/png/zopflipng.cc]: Removing node.h breaks some linkers 

see : https://groups.google.com/d/msg/nodejs/onA0S01INtw/ZnfKyN7sGAsJ

**Comment by pierreinglebert** [comment]
@jakubpawlowicz : it has been a bit longer than I thought :)

**Comment by jakubpawlowicz** [comment]
No worries, thanks for taking care of this module :+1:

**Pull request merged**